enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -730,6 +725,19 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 
 		log.Debugf("PublishOperation: operation %q stage: %s", taskID, stage)
 
+		// Before publishing the COMPLETED operation, update the task router
+		// so that subsequent tasks with similar routing properties can get
+		// routed back to the same executor.
+		if stage == repb.ExecutionStage_COMPLETED {
+			response := operation.ExtractExecuteResponse(op)
+			if response != nil {
+				if err := s.updateRouter(ctx, taskID, response); err != nil {
+					// Errors from updating the router should not be fatal.
+					log.Errorf("Failed to update task router: %s", err)
```

Comment:
Can you add the taskID to the error message (unless the returned error already includes it)

Reply:
Done